src/pages/receive.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react'
+import { Box, Flex, FormControl, FormLabel, FormErrorMessage, Button, Input, Heading, Stack, Text } from '@chakra-ui/core'
+import { useForm } from "react-hook-form";
+import { Redirect } from 'react-router-dom';
+import { useMutation, gql } from '@apollo/client';
+import { AddressInput } from '../generated/graphql';
+
+type RecipientFormData = {
+    name: string;
+    password: string;
+    passwordConfirmation: string
+    address: AddressInput
+    instructions: String
+};
+
+const CREATE_RECIPIENT = gql`
+    mutation CreateRecipient($name: String!, $password: String!, $address: AddressInput!, $instructions: String) {
+        createRecipient(name: $name, password: $password, address: $address, instructions: $instructions) {
+            did
+            name
+            address
+            instructions
+        }
+    }
+`
+
+export function ReceivePage() {
+    const { handleSubmit, errors, setError, register, formState } = useForm<RecipientFormData>();
+
+    const [createRecipient,] = useMutation(CREATE_RECIPIENT)
+    const [wasAdded, setWasAdded] = useState(false)
+
+    async function onSubmit({ name, password, passwordConfirmation, address, instructions }: RecipientFormData) {
+        if (password !== passwordConfirmation) {
+            setError("password", "invalid", "password confirmation doesn't match")
+            return
+        }
+        try {
+            await createRecipient({ variables: { name: name, password: password, address: address, instructions: instructions } })
+        } catch (e) {
+            setError("name", "unknown", `Could not create that donation recipient: ${e.message}`)
+            return
+        }
+        setWasAdded(true)
```

Comment:
I think this can just be a history.push() here instead of the state and the Redirect